features/support/openhab.rb

```diff
@@ -177,14 +177,23 @@ def delete_things
   openhab_client('openhab:things clear')
 end
 
-def delete_rules
+def delete_rules # rubocop:disable Metrics/MethodLength
   FileUtils.rm Dir.glob(File.join(rules_dir, '*.rb'))
   deleted = false
-  Rest.rules.each do |rule|
-    uid = rule['uid']
-    Rest.delete_rule(uid)
-    deleted = true
+  begin
+    retries ||= 0
+    Rest.rules.each do |rule|
+      uid = rule['uid']
+      Rest.delete_rule(uid)
+      deleted = true
+    end
+  rescue StandardError => e
+    raise e unless (retries += 1) < 3
```

Comment:
You should just `raise` not `raise e`. The latter resets the backtrace as if the exception is thrown from this point, whereas the former preserves the original backtrace.